src/ucp/tag/eager_rcv.c

```diff
@@ -13,6 +13,58 @@
 #include <ucs/datastruct/queue.h>
 #include <ucp/core/ucp_request.inl>
 
+static UCS_F_ALWAYS_INLINE void
+ucp_eager_expected_handler(ucp_worker_t *worker, ucp_request_t *req,
+                           void *data, size_t recv_len, ucp_tag_t recv_tag,
+                           uint16_t flags)
+{
+    ucs_trace_req("found req %p", req);
+    UCS_PROFILE_REQUEST_EVENT(req, "eager_recv", recv_len);
+
+    /* First fragment fills the receive information */
+    UCP_WORKER_STAT_EAGER_MSG(worker, flags);
+    UCP_WORKER_STAT_EAGER_CHUNK(worker, EXP);
+
+    req->recv.tag.info.sender_tag = recv_tag;
+
+    /* Cancel req in transport if it was offloaded,
+     * because it arrived either:
+     * 1) via SW TM (e. g. peer doesn't support offload)
+     * 2) as unexpected via HW TM */
+    ucp_tag_offload_try_cancel(worker, req,
+                               UCP_TAG_OFFLOAD_CANCEL_FORCE |
+                               UCP_TAG_OFFLOAD_CANCEL_DEREG);
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_eager_offload_handler(void *arg, void *data, size_t length,
+                          unsigned tl_flags, uint16_t flags, ucp_tag_t recv_tag)
+{
+    ucp_worker_t *worker = arg;
+    ucp_request_t *req;
+    ucp_recv_desc_t *rdesc;
+    ucp_tag_t *rdesc_hdr;
+    ucs_status_t status;
+
+    req = ucp_tag_exp_search(&worker->tm, recv_tag);
+    if (req != NULL) {
+        ucp_eager_expected_handler(worker, req, data, length, recv_tag, flags);
+        req->recv.tag.info.length = length;
```

Comment:
perfectionism: alignment of ```=``` on this and next lines

Reply:
thought about this, but then the following lines would not fit to 80 sym limit